### PR TITLE
Add "WSL + Virtual Environment + VSCode" section to Setting Up docs

### PR DESCRIPTION
**Pull request by adityagandhamal**
Closes [this card](https://trello.com/c/Jy58U3tA/813-setting-up-wsl-virtual-environment-vscode).

### CODE REVIEW

**Comment by V-G-spec** [comment]
Looks good to me, but unsure about the functionality. @zaeemansari70 can you take a look as well, and if all's good then mark the corresponding Trello card as done?
Thanks both :D

**Comment by adityagandhamal** [comment]
Cc: @KareemMAX

File: docs/overview/contributing/setting_up.rst [hunk 1]
```diff
@@ -694,6 +694,69 @@ Note: Currently you do not need to comment out the :code:`conftest.py` file in t
     </iframe>
 
 
+Setting up with WSL
+-------------------
+
+Here is another way for you to set up the whole Ivy environment on Windows using WSL, virtual environment and VSCode.
```
Comment (zaeemansari70) [inline docs/overview/contributing/setting_up.rst:700]: Also please remove this first line, we don't need to specify this type of stuff here. :slightly_smiling_face: 
`
Here is another way for you to set up the whole Ivy environment on Windows using WSL, virtual environment and VSCode. `

**Review by zaeemansari70 — CHANGES_REQUESTED** [review]
Hi @adityagandhamal ,

This is not related to `Codespaces`, you have added it under the `Codespaces` section.
This should be under the `Setting up for Free` section https://github.com/unifyai/ivy/blob/3aa0df60585f6eb35250b51864a432a3eb760d71/docs/overview/contributing/setting_up.rst?plain=1#L526 , and after the `Windows` and `Ubuntu` subsection, with the heading just named as `WSL`. 
After this subheading here :slightly_smiling_face:  https://unify.ai/docs/ivy/overview/contributing/setting_up.html#id8

Do let me know if you have any questions :slightly_smiling_face: 
Thanks!

**Comment by adityagandhamal** [comment]
thanks for the review @zaeemansari70.
Absolutely, this is not related to the Codespaces at all and I have added this as a completely new section at the end, making sure it does not fall under any of the previous sections. The reason being:
1. Could be added under `Virtual environments - No Docker` but it discusses both PyCharm and VSCode, unlike the new section where we only stick to VSCode.

2. The `Setting Up for Free` section discusses docker thus contradicting the new section.

3. Also, IMO, the users would find it appealing to have an entire dedicated WSL guide.

Interested to know your thoughts on this!

**Review by zaeemansari70 — CHANGES_REQUESTED** [review]
Hi @adityagandhamal , 

Thanks for explaining why you added it here, it makes sense that you are thinking of adding it as a separate heading. 

Actually the main reason would be that this is somewhat a perfect guide for a `setting up for free` section, as we are using `WSL` as well as `vscode` it is also way lighter on systems as well as free, also those sections are not at all dedicated specifically for `docker` thise are paid too(setting up an IDE with a trial account stuff).
Moreover, if it was a completely separate distribution it would've made sense to add it in a completely separate sections :slightly_smiling_face: 

Do let me know if you have any other questions!
Thanks! :slightly_smiling_face:

**Comment by adityagandhamal** [comment]
Fair enough, your point is valid. Though a lightweight setup, my thought process should have been inclined towards the free side of the stuff here. 
I will make the necessary changes accordingly.
Thank you @zaeemansari70

**Review by zaeemansari70 — APPROVED** [review]
@adityagandhamal,

Thanks for the changes! :slightly_smiling_face:

**Pull request merged**